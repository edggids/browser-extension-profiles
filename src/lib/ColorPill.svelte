<script lang="ts">
import {createEventDispatcher} from "svelte";
import type {ColorModel} from "../models/ColorModel";

export let color:ColorModel;
export let label:string|null = null;

let className = '';
export { className as class };

const dispatch = createEventDispatcher();

$:wrapperStyle = `--color-current-light: ${color.light}`;
$:wrapperClass = `inline-flex items-center p-2 hover:bg-current-light ${className}`;

function onClick() { dispatch('click'); }
</script>

<span role="button" on:click={onClick} class={wrapperClass} style={wrapperStyle}>
    <span class="inline-flex mr-3">
        <span style="width: 16px; height:24px; background-color: {color.light}"></span>
        <span style="width: 16px; height:24px; background-color: {color.default}"></span>
        <span style="width: 16px; height:24px; background-color: {color.darker}"></span>
    </span>

    {#if label}
        <span class="text leading-none pr-3">{label}</span>
    {/if}
</span>
