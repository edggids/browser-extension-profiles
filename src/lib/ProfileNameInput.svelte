<script lang="ts">
    let className = '';
    export { className as class };

    export let editable = false;
    export let value = '';
</script>

<div data-testid="inputGroup" class={`container transition-bg bg-current-light transition-color text-current text-lg pt-2 pb-3.5 ${className}`}>
    <span class="text-xl">{'{'}</span>
    {#if editable}
        <input placeholder="SET NAME HERE" class="plain" type="text" bind:value />
    {:else}
        <span>{ value }</span>
    {/if}
    <span class="text-xl">{'}'}</span>
</div>
