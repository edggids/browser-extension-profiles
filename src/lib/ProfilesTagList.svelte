<script lang="ts">
    import Tag from "./Tag.svelte";
    import {createEventDispatcher} from "svelte";

    export let profiles = [];
    export let currentProfileId;

    const dispatch = createEventDispatcher();
    const onClick = (id) => dispatch("selected", { id });
</script>

{#each profiles as [id, profile]}
    <Tag active={id === currentProfileId } on:click={e => onClick(id)} class="mr-4 mb-4" color={profile.color.default}>{profile.name}</Tag>
{/each}
