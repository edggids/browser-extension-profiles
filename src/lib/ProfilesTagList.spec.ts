import {cleanup, render} from '@testing-library/svelte';
import {describe, it, expect, afterEach, vi} from 'vitest';
import ProfilesTagList from "./ProfilesTagList.svelte";
import ColorPill from "./ColorPill.svelte";
import ProfileNameInput from "./ProfileNameInput.svelte";

describe('ProfilesTagList component', () => {
  afterEach(() => {
    cleanup();
  });

  it('mounts', () => {
    const { container } = render(ProfilesTagList, { props: { profiles: [], currentProfileId: 'id1' } });

    expect(container).toMatchSnapshot();
  });

  it('renders no tags when no profiles are provided', () => {
    const { container } = render(ProfilesTagList, { props: { profiles: [], currentProfileId: 'id1' } });
    expect(container.firstChild.childNodes.length).toBe(1);
  });

  it('renders tags for provided profiles', () => {
    const profiles = [['id1', { name: 'Profile 1', color: { default: '#123456' } }], ['id2', { name: 'Profile 2', color: { default: '#654321' } }]];
    const { container } = render(ProfilesTagList, { props: { profiles, currentProfileId: 'id1'} });

    expect(container.firstChild.childNodes.length).toBe(profiles.length + 1);
  });

  it('dispatches click event', async () => {
    const profiles = [['id1', { name: 'Profile 1', color: { default: '#123456' } }], ['id2', { name: 'Profile 2', color: { default: '#654321' } }]];
    const { container, component } = render(ProfilesTagList, { props: { profiles, currentProfileId: 'id1'} });
    const tag = container.firstChild.firstChild;

    const onClick = vi.fn();
    component.$on('selected', onClick)

    tag.dispatchEvent(new MouseEvent('click'));
    expect(onClick).toHaveBeenCalled();
  })
});
