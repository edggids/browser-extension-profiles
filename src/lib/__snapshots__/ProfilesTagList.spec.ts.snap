// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfilesTagList component > mounts 1`] = `
<body>
  <div>
    
  </div>
</body>
`;
